app/controllers/settings_controller.rb

```diff
@@ -4,63 +4,60 @@ class SettingsController < ApplicationController
     controller.ensure_logged_in t("layouts.notifications.you_must_log_in_to_view_your_settings")
   end
 
-  before_filter :except => :unsubscribe do |controller|
-    controller.ensure_authorized t("layouts.notifications.you_are_not_authorized_to_view_this_content")
-  end
+  before_filter EnsureCanAccessPerson.new(:person_id, error_message_key: "layouts.notifications.you_are_not_authorized_to_view_this_content"), except: :unsubscribe
 
   def show
+    target_user = Person.find_by_username_and_community_id(params[:person_id], @current_community.id)
+    target_user = add_location_to_person(target_user)
```

Comment:
This could be changed to `add_location_to_person!(target_user)` and not return the person, instead mutate it in the method.